File: pbkdpwls/구현/귤 고르기.py
from collections import Counter


def solution(k, tangerine):
    answer = 0

    dic1 = dict(sorted(Counter(tangerine).items(), key=lambda x: x[1], reverse=True))

    count = 0
    for i in dic1.values():
        if (count >= k):
            break
        count += i
        answer += 1
    return answer

File: yunva17/프로그래머스/Lv2/귤_고르기.js
const solution = (k, tangerine) => {
  let answer = 0;
  let fruits = {};

  tangerine.forEach((v) => {
    if (!fruits[v]) {
      fruits[v] = 1;
    } else {
      fruits[v]++;
    }
  });

  fruits = Object.entries(fruits).sort((a, b) => b[1] - a[1]);

  fruits.forEach((v, i) => {
    if (k > 0) {
      k = k - v[1];
      answer++;
    } else {
      return;
    }
  });

  return answer;
};
